app/bundles/LeadBundle/Tests/Functional/EventListener/SegmentSubscriberTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\LeadBundle\Tests\Functional\EventListener;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\LeadBundle\Entity\LeadList;
+use Mautic\LeadBundle\Entity\LeadListRepository;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class SegmentSubscriberTest extends MauticMysqlTestCase
+{
+    /**
+     * @dataProvider filterProvider
+     *
+     * @param mixed[]  $filters
+     * @param string[] $expectedTranslations
+     */
+    public function testSegmentFilterAlertMessages(array $filters, array $expectedTranslations): void
+    {
+        $segment   = $this->saveSegment('Segment D', 'segment-d', $filters);
+        $crawler   = $this->client->request(Request::METHOD_GET, '/s/segments/edit/'.$segment->getId());
+        Assert::assertTrue($this->client->getResponse()->isOk());
+        /** @var TranslatorInterface $translator */
+        $translator = $this->getContainer()->get('translator');
+
+        $expectedTranslationString = implode(' ', array_map(function ($trans) use ($translator) {
+            return $translator->trans($trans);
+        }, $expectedTranslations));
```

Comment:
One thing about arrow functions is that it doesn't need the `use()` part. It saves like 34 characters, so this is more of a micro optimization, but I find it more readable. It may be a personal preference though.
```suggestion
        $expectedTranslationString = implode(' ', array_map(fn ($trans) => $translator->trans($trans), $expectedTranslations));
```